sinker/sinker.go

```diff
@@ -68,14 +71,18 @@ func (s *SQLSinker) Run(ctx context.Context) {
 		}
 	}
 
-	s.Sinker.OnTerminating(s.Shutdown)
-	s.OnTerminating(func(err error) {
+	s.Sinker.OnTerminating(func(err error) {
+		if err == nil {
+			s.logger.Info("sql sinker terminating, flushing remaining rows")
+			err = s.flush(ctx)
+		}
 		s.stats.LogNow()
-		s.logger.Info("sql sinker terminating", zap.Stringer("last_block_written", s.stats.lastBlock))
-		s.Sinker.Shutdown(err)
+		s.logger.Info("sql sinker terminated", zap.Stringer("last_block_written", s.stats.lastBlock), zap.Error(err))
+		s.Shutdown(err)
 	})
+	s.OnTerminating(s.Sinker.Shutdown)
 
-	s.OnTerminating(func(_ error) { s.stats.Close() })
```

Comment:
It's when `s` terminates that we want to close the stats. It feels weird that stats is registering its own Terminating handler to close itself.